python/cuml/common/base.pyx

```diff
@@ -364,6 +368,24 @@ class Base(TagsMixin,
             return {'preserves_dtype': [self.dtype]}
         return {}
 
+    def _set_nvtx_annotations(self):
```

Comment:
Do we need as a separate method as opposed to being part of the init?

Reply:
This allows to enable NVTX benchmark either by setting the environment variable or otherwise by calling the `set_nvtx_annotations` method on the estimators we want to follow.

Reply:
Actually, yes I can move the code of `_set_nvtx_annotations` specifically, and put it into the `__init__` function. Thanks for the recommendation.